app/models/assignment/rotation_generator.rb

```diff
@@ -0,0 +1,60 @@
+# frozen_string_literal: true
+
+class Assignment < ApplicationRecord
+  class RotationGenerator
+    include ActiveModel::Model
+    include ActiveModel::Attributes
+
+    attribute :roster_id, :integer
+    attribute :user_ids
+    attribute :starting_user_id, :integer
+    attribute :start_date, :date
+    attribute :end_date, :date
+
+    validates :roster, presence: true
+    validates :user_ids, presence: true, length: { minimum: 1 }
+    validate :includes_start_user
+    validates :start_date, presence: true
+    validates :end_date, presence: true, comparison: { greater_than_or_equal_to: :start_date }
```

Comment:
```suggestion
    validates :roster, presence: true
    validates :user_ids, presence: true
    validates :starting_user_id, presence: true
    validates :start_date, presence: true
    validates :end_date, presence: true, comparison: { greater_than_or_equal_to: :start_date }
    validate :includes_start_user
```
Couple of things:
- `presence: true` should cover the `minimum: 1` condition since `[].present? => false`
- validate that `starting_user_id` is present separately
- keep custom validation declarations separate from built-in ones